### DISCUSSION TITLE
API 들을 한 파일에 정의하는건 어떠신가요?

### DISCUSSION THREAD

**Discussion by aLVINlEE9** [GhostPangPang/GhostPong#264] [General]
API 들을 한 파일에 정의하는건 어떠신가요?

```typescript
import { post, patch } from '@/libs/api';
import { AxiosResponse } from 'axios';

const API = '/user/image';

const postFileUpload = async (formData: FormData): Promise<AxiosResponse> => {
  return await post<AxiosResponse>(API, formData);
};

const patchFileUpload = async (location: string): Promise<AxiosResponse> => {
  return await patch<AxiosResponse>(API, { image: location });
};
```

이런식으로 항상 API와 dto(entity) 들을 가져와서 HOOK을 만들어야하기 때문에 이를 한 파일에 정의하는것은 어떠신가요?
@nyeoni

**Comment by nyeoni** [comment]
`api` 디렉토리를 파서 `postFileUpload` 등의 함수를 넣자는 말씀이신가요?
만약 그렇다면, 저는 오히려 파일을 수정해야할때 파일 간 이동이 있어서 한 파일에서 관리하는게 좋다고 생각합니다! 제가 이해한 내용이 맞는지 확인해주세용!

**Reply by aLVINlEE9** [reply]
네 그런데 api 명세가 정해진 이상 api를 호출하는 `postFileUpload`와 같은 함수들은 변할 가능성이 높지않다고 봅니다. 어떻게 생각하시나요?

**Reply by nyeoni** [reply]
변할 가능성은 낮지만, 변하지 않는다고 뺄 필요성은 못느끼는 것 같습니다! 저희는 api hook 을 따로 만들었기 때문에, 해당 fetch 함수 (ex. useUser hook - getUser) 는 훅 외에서는 쓰이지 않는데, 따로 구분해서 얻는 장점을 잘 모르겠습니다ㅜㅜ
